pkg/cloudprovider/suite_test.go

```diff
@@ -280,6 +282,23 @@ var _ = Describe("CloudProvider", func() {
 				Expect(err).ToNot(HaveOccurred())
 				Expect(drifted).To(Equal(ImageDrift))
 			})
+
+			It("should NOT trigger drift if node doesn't have kubelet client ID label", func() {
```

Comment:
Have UTs but don't have an E2E. It is probably possible to write an E2E but it would require:
1. Provisioning a new Kubelet identity (calling ARM APIs, which I don't think we currently import the SDKs to do this, though it's not that hard to do).
2. Using the `ExpectSettingsReplaced` helper (thanks @Bryce-Soghigian for pointing me to it), which can't be tested in NAP.

It's not clear to me if we should add an E2E for this or if we'd prefer just UTs.